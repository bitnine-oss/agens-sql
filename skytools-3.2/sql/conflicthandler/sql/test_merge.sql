
\set ECHO none
\i merge_on_time.sql
\set ECHO all

set DateStyle='ISO';

create table mergetest (
    intcol int4,
    txtcol text,
    timecol timestamp
);

-- insert to empty
select merge_on_time('timefield=timecol', null, null, null, null, null, 'I:intcol', 'intcol=5&txtcol=v1&timecol=2010-09-09+12:12', 'mergetest', null, null, null);
select * from mergetest;

-- insert to with time earlier
select merge_on_time('timefield=timecol', null, null, null, null, null, 'I:intcol', 'intcol=5&txtcol=v2&timecol=2010-09-08+12:12', 'mergetest', null, null, null);
select * from mergetest;

-- insert to with time later
select merge_on_time('timefield=timecol', null, null, null, null, null, 'I:intcol', 'intcol=5&txtcol=v3&timecol=2010-09-10+12:12', 'mergetest', null, null, null);
select * from mergetest;

