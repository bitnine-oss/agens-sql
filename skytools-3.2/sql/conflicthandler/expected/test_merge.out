\set ECHO none
set DateStyle='ISO';
create table mergetest (
    intcol int4,
    txtcol text,
    timecol timestamp
);
-- insert to empty
select merge_on_time('timefield=timecol', null, null, null, null, null, 'I:intcol', 'intcol=5&txtcol=v1&timecol=2010-09-09+12:12', 'mergetest', null, null, null);
    merge_on_time    
---------------------
 data ok, no old row
(1 row)

select * from mergetest;
 intcol | txtcol |       timecol       
--------+--------+---------------------
      5 | v1     | 2010-09-09 12:12:00
(1 row)

-- insert to with time earlier
select merge_on_time('timefield=timecol', null, null, null, null, null, 'I:intcol', 'intcol=5&txtcol=v2&timecol=2010-09-08+12:12', 'mergetest', null, null, null);
                   merge_on_time                   
---------------------------------------------------
 IGN:data ok, old row, current row more up-to-date
(1 row)

select * from mergetest;
 intcol | txtcol |       timecol       
--------+--------+---------------------
      5 | v1     | 2010-09-09 12:12:00
(1 row)

-- insert to with time later
select merge_on_time('timefield=timecol', null, null, null, null, null, 'I:intcol', 'intcol=5&txtcol=v3&timecol=2010-09-10+12:12', 'mergetest', null, null, null);
          merge_on_time           
----------------------------------
 data ok, old row, new row better
(1 row)

select * from mergetest;
 intcol | txtcol |       timecol       
--------+--------+---------------------
      5 | v3     | 2010-09-10 12:12:00
(1 row)

